corehq/apps/products/fixtures.py

```diff
@@ -52,8 +53,19 @@ def get_products():
                 key=lambda product: product.code
             )
 
-        return simple_fixture_generator(
+        fixture_nodes = simple_fixture_generator(
             restore_user, self.id, "product", PRODUCT_FIELDS, get_products, restore_state.last_sync_log
         )
+        if not fixture_nodes:
+            return []
+
+        if restore_state.params.openrosa_version and restore_state.params.openrosa_version < '2.1':
+            # Don't include index schema when openrosa version is specified and below 2.1
```

Comment:
I might be missing something very obvious from your code, but is there a reason you didn't do the inverse:
```py
if LooseVersion(openrosa_version) >= LooseVersion(OPENROSA_VERSION_MAP['INDEXED_PRODUCTS']:
    schema_node = get_index_schema_node(self.id, ['@id', 'code', 'program_id', 'category'])
    fixture_nodes[0].attrib['indexed'] = 'true'
    return [schema_node] + fixture_nodes
else: 
    return fixture_nodes
```
That way you would have maintained the convention, and we will retain a map of "features to openrosa versions" in the constant. 

Also, then you can keep passing the default version around when this isn't specified (although I don't really care about that as much).